crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -185,6 +185,14 @@ impl Persistence {
             .ok_or(error::Auction::Missing)
             .map(|scores| (scores.block_deadline as u64).into())?;
 
+        if database::settlements::already_processed(&mut ex, auction_id)
+            .await
+            .context("fetch already_processed")
+            .map_err(error::Auction::DbError)?
+        {
+            return Err(error::Auction::AlreadyProcessed);
+        }
+
```

Comment:
Would you prefer for this check to be outside of the `get_auction` so that `get_auction` is more reusable? With this check I feel we are embedding `OnSettlementEventUpdater` requirements into `get_auction`?

Reply:
Yes, I think there should be `get_auction` and `get_unprocessed_auction` which calls `get_auction` and additionally checks that it's not yet processed.

Reply:
Added a separate function `infra::persistence::auction_has_settlement` that solely checks for this edge case, and once we eliminate the edge case we can just remove the function and not touch rest of the code.